Refresh.Interfaces.APIv3/Endpoints/CommentApiEndpoints.cs

```diff
@@ -0,0 +1,170 @@
+
+        return ApiProfileCommentResponse.FromOld(comment, dataContext);
+    }
+    
+    [ApiV3Endpoint("profileComments/id/{id}", HttpMethods.Delete)]
+    [DocSummary("Deletes the profile comment specified by it's ID. Fails if the user is not the comment poster or the profile owner.")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.CommentMissingErrorWhen)]
+    [DocError(typeof(ApiValidationError), ApiValidationError.NoCommentDeletionPermissionErrorWhen)]
+    public ApiOkResponse DeleteProfileComment(RequestContext context, DataContext dataContext, GameUser user, int id)
+    {
+        GameProfileComment? comment = dataContext.Database.GetProfileCommentById(id);
+        if (comment == null) return ApiNotFoundError.CommentMissingError;
+
+        if (user != comment.Author && user != comment.Profile) return ApiValidationError.NoCommentDeletionPermissionError;
+
+        dataContext.Database.DeleteProfileComment(comment);
+        return new ApiOkResponse();
+    }
+
+    [ApiV3Endpoint("profileComments/id/{id}/rate/{rawRating}", HttpMethods.Post)]
+    [DocSummary("Rates the profile comment specified by it's ID.")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.CommentMissingErrorWhen)]
+    [DocError(typeof(ApiValidationError), ApiValidationError.RatingParseErrorWhen)]
+    public ApiOkResponse RateProfileComment(RequestContext context, DataContext dataContext, GameUser user, int id, 
+        [DocSummary("The user's new rating for the comment. -1 = dislike, 0 = neutral, 1 = like.")] string rawRating)
+    {
+        GameProfileComment? comment = dataContext.Database.GetProfileCommentById(id);
+        if (comment == null) return ApiNotFoundError.CommentMissingError;
+
+        // rawRating is string and not sbyte or integer because passing any out of range value will make Bunkum 
+        // set rawRating to 0 instead, which we would here wrongly take as a neutral rating instead of an invalid value.
+        if (!sbyte.TryParse(rawRating, out sbyte rating) || !Enum.IsDefined(typeof(RatingType), rating))
+            return ApiValidationError.RatingParseError;
+
+        dataContext.Database.RateProfileComment(user, comment, (RatingType)rating);
+        return new ApiOkResponse();
+    }
+    #endregion
+
+    #region Level
+    [ApiV3Endpoint("levels/id/{id}/comments"), Authentication(false)]
+    [DocSummary("Gets comments posted under the specified level.")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.LevelMissingErrorWhen)]
+    [DocUsesPageData]
+    public ApiListResponse<ApiLevelCommentResponse> GetCommentsOnLevel(RequestContext context, DataContext dataContext, int id)
+    {
+        GameLevel? level = dataContext.Database.GetLevelById(id);
+        if (level == null) return ApiNotFoundError.LevelMissingError;
+
+        (int skip, int count) = context.GetPageData();
+
+        DatabaseList<GameLevelComment>? comments = dataContext.Database.GetLevelComments(level, count, skip);
```

Comment:
These shouldn't ever be null
```suggestion
        DatabaseList<GameLevelComment> comments = dataContext.Database.GetLevelComments(level, count, skip);
```